pkg/controller/migration/controller.go

```diff
@@ -0,0 +1,65 @@
+package migration
+
+import (
+	"github.com/go-kit/kit/log"
+	api_v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/sapcc/kubernikus/pkg/apis/kubernikus/v1"
+	"github.com/sapcc/kubernikus/pkg/controller/base"
+	"github.com/sapcc/kubernikus/pkg/controller/config"
+	"github.com/sapcc/kubernikus/pkg/migration"
+	"github.com/sapcc/kubernikus/pkg/util"
+)
+
+const (
+	MigrationFailed = "MigrationFailed"
+)
+
+type MigrationReconciler struct {
+	config.Clients
+
+	Recorder record.EventRecorder
+	Logger   log.Logger
+}
+
+func NewController(threadiness int, factories config.Factories, clients config.Clients, recorder record.EventRecorder, logger log.Logger) base.Controller {
+	logger = log.With(logger,
+		"controller", "migration")
+
+	var reconciler base.Reconciler
+	reconciler = &MigrationReconciler{clients, recorder, logger}
+
+	return base.NewController(threadiness, factories, reconciler, logger, nil)
+}
+
+func (mr *MigrationReconciler) Reconcile(kluster *v1.Kluster) (bool, error) {
+
+	//We only care about klustes with pending migrations
+	if !migration.MigrationsPending(kluster) {
```

Comment:
These too loops are a not obvious. Double negation and noop in the called function. Feels like it could be unraveled.

Reply:
I'm not sure I follow. Care to elaborate?